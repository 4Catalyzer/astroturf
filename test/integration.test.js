/**
 * @jest-environment node
 */

import fs from 'fs';
import ExtractCSS from 'mini-css-extract-plugin';
import webpack from 'webpack';

const outputPath = `${__dirname}/build`;
describe('webpack integration', () => {
  let compiler;

  beforeEach(() => {
    compiler = webpack({
      devtool: false,
      mode: 'development',
      entry: require.resolve('./integration'),
      output: {
        path: outputPath,
      },
      module: {
        rules: [
          {
            test: /\.css$/,
            use: [
              ExtractCSS.loader,
              // require.resolve('../src/css-loader'),
              {
                loader: 'css-loader',
                options: { modules: true, importLoaders: 1 },
              },
              {
                loader: 'postcss-loader',
                options: {
                  ident: 'postcss-astroturf',
                  plugins: [require('postcss-nested')()], // eslint-disable-line global-require
                },
              },
            ],
          },
          {
            test: /\.jsx?$/,
            use: ['babel-loader', require.resolve('../src/loader')],
          },
        ],
      },
      resolve: {
        alias: {
          astroturf: require.resolve('../src/runtime/styled'),
        },
      },
      plugins: [new ExtractCSS()],
    });
  });

  it('should work', done => {
    compiler.run((err, stats) => {
      console.log(stats.toJson().errors);
      expect(err).toBe(null);
      expect(stats.hasErrors()).toBe(false);
      expect(stats.hasWarnings()).toBe(false);

      expect(
        fs.readFileSync(`${outputPath}/main.css`, 'utf8'),
      ).toMatchSnapshot();

      done();
    });
  });
});
