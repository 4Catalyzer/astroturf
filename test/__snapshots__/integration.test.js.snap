// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`webpack integration should work 1`] = `
"
  .index-styles__parent {
    color: red;
  }

.index-FancyBox__cls1 {
color: red;
width: 75px;
}

.index-FancyBox__cls1.index-FancyBox__primary {
background: white;
color: palevioletred;
}
.index-FancyBox__cls2 {
}
.index-FancierBox__cls1 {
color: ultra-red;
}


.index-FancierBox__cls1 > .index-FancyBox__cls1 {
padding: 4em;
}


.index-styles__parent > .index-FancierBox__cls1 {
margin: 2em;
}
.index-Button__cls1 {
/*!*/
}
.index-Button__cls2 {
}
"
`;
